bindata/network/ovn-kubernetes/common/008-script-lib.yaml

```diff
@@ -0,0 +1,547 @@
+        --db-nb-sock=${nbdb_sock} \
+        --ovn-nb-log="-vconsole:${log_level} -vfile:off -vPATTERN:console:{{.OVN_LOG_PATTERN_CONSOLE}}" \
+        run_nb_ovsdb &
+      wait $!
+    }
+
+    # retry() an operation a number of times, sleeping 2 seconds between each try
+    retry() {
+      local tries=${1}
+      local desc=${2}
+      local cmd=${3}
+
+      local retries=0
+      while ! ${cmd}; do
+        (( retries += 1 ))
+        if [[ "${retries}" -gt ${tries} ]]; then
+          echo "$(date -Iseconds) - ERROR - nbdb ${desc} - too many failed attempts, giving up"
+          return 1
+        fi
+        echo "$(date -Iseconds) - WARN - nbdb ${desc} - failed try ${retries}, retrying..."
+        sleep 2
+      done
+      echo "$(date -Iseconds) - INFO - nbdb ${desc} - success"
+      return 0
+    }
+
+    # nbdb-post-start() tweaks nbdb database server settings and sets a number
+    # of options in NB_Globals to configure OVN global settings
+    nbdb-post-start()
+    {
+      local northd_probe_interval=${1:-10000}
+
+      rm -f ${nbdb_pidfile}
+
+      # set inactivity probe
+      if ! retry 60 "inactivity-probe" "ovn-nbctl -t 5 --inactivity-probe={{.OVN_NB_INACTIVITY_PROBE}} set-connection punix:${nbdb_sock}"; then
+        exit 1
+      fi
+      # set trim-on-compaction
+      if ! retry 60 "trim-on-compaction" "ovn-appctl -t ${nbdb_ctl} --timeout=5 ovsdb-server/memory-trim-on-compaction on"; then
+        exit 1
+      fi
+
+      # set IC zone
+      echo "Setting the IC zone to ${K8S_NODE}"
+      IC_OPTION="name=\"${K8S_NODE}\" options:name=\"${K8S_NODE}\""
+
+      # northd probe interval
+      echo "Setting northd probe interval to ${northd_probe_interval} ms"
+      NORTHD_PROBE_OPTION="options:northd_probe_interval=${northd_probe_interval}"
+
+      # let northd sleep so it takes less CPU
+      NORTHD_SLEEP_OPTION="options:northd-backoff-interval-ms={{.OVN_NORTHD_BACKOFF_MS}}"
+
+      local ipsec=false
+      local ipsec_encapsulation=false
+{{ if .OVNIPsecEnable }}
+      ipsec=true
+      # IBMCloud does not forward ESP (IP proto 50)
+      # Instead, force IBMCloud IPsec to always use NAT-T
+      if [ "{{.PlatformType}}" == "IBMCloud" ]; then
+        ipsec_encapsulation=true
+      fi
+{{ end }}
+      IPSEC_OPTION="ipsec=${ipsec} options:ipsec_encapsulation=${ipsec_encapsulation}"
+
+      # set all the NB_GLOBAL options
+      if ! retry 20 "nb-global options" "ovn-nbctl -t 5 set nb_global . ${IC_OPTION} ${NORTHD_PROBE_OPTION} ${NORTHD_SLEEP_OPTION} ${IPSEC_OPTION}"; then
+        exit 1
+      fi
+    }
+
+    # ovndb-readiness-probe() checks if the the database is in the active state
+    # and if not, exits with an error code.
+    ovndb-readiness-probe()
+    {
+      # dbname should be 'sb' or 'nb'
+      local dbname=$1
+
+      local ctlfile
+      if [[ "${dbname}" = "sb" ]]; then
+        ctlfile=${nbdb_ctl}
+      elif [[ "${dbname}" = "nb" ]]; then
+        ctlfile=${sbdb_ctl}
```

Comment:
Just a heads up that you started using this variable in a commit but defined it in a later commit.

Reply:
@jcaamano hmm, can you elaborate? do you mean `dbname` or `ctlfile`?  AFAICS I define and use both of those in `ovnkube: move nbdb container script to common library` but I'm sure I'm missing something

Reply:
Sorry, wasn't clear enough here.

I meant `sbdb_ctl` which started to be used in `ovnkube: move nbdb container script to common library` but was defined in `ovnkube: move sbdb container script to common library``